modules/misc/editorconfig.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.editorconfig;
+
+  iniFormat = pkgs.formats.ini { };
+
+in {
+  meta.maintainers = with maintainers; [ loicreynier ];
+
+  options.editorconfig = {
+    enable = mkEnableOption "EditorConfig home configuration file";
+
+    settings = mkOption {
+      type = iniFormat.type;
+      default = { };
+      description = ''
+        Configuration written to <filename>$HOME/.editorconfig</filename>.
+        <code>root = true</code> is automatically added to the file,
+        it must no be added in here.
+        See <link xlink:href="https://editorconfig.org"/> for documentation.
+      '';
+      example = literalExpression ''
+        {
+          "*" = {
+            charset = "utf-8";
+            end_of_line = "lf";
+            trim_trailing_whitespace = true;
+            insert_final_newline = true;
+            max_line_width = 78;
+            indent_style = "space";
+            indent_size = 4;
+          };
+        };
+      '';
+    };
+  };
+
+  config = mkIf (cfg.enable && cfg.settings != { }) {
+    home.file.".editorconfig".text = let
+      renderedSettings = generators.toINIWithGlobalSection { } {
+        globalSection = { root = true; };
```

Comment:
We should provide a way for the user to add things to the global section.

Reply:
Oh, I didn't realize that there was only one thing that could go in the global section. I guess you use `[*]` for other settings.

It may be worth providing a way to make it non-root, but I am not quite sure about the utility there, so this is probably fine

Reply:
Exactly, all settings other than `root = true` are in `[*]`.

I added a `rootConfig` option to provide a way to make it non-root. Not sure about the name though. Is there a naming convention for this kind of option? Maybe `enableRootConfig`?

Reply:
I'd say to just leave it hard coded as `root = true` (i.e., not introduce an option for it). I doubt anybody would need to set it to `false` for the `~/.editorconfig` file. If somebody really wants it they can open an issue and we can worry about it then.